### PR TITLE
Add support for the aws_cognito_user_pool_domain resource

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -201,3 +201,11 @@ resource "aws_cognito_user_pool_client" "client" {
   user_pool_id                         = aws_cognito_user_pool.user_pool[0].id
   write_attributes                     = each.value.write_attributes
 }
+
+resource "aws_cognito_user_pool_domain" "domain" {
+  count = var.module_enabled && length(var.domain_prefix) > 0 ? 1 : 0
+
+  domain          = var.domain_prefix
```
Comment (mariux) [inline main.tf:208]: should not be `domain_prefix` but `domain`

File: README.md [hunk 2]
```diff
@@ -350,6 +349,20 @@ for details and use-cases.
...
   }
   ```
+#### Cognito User Pool Domain
+
+- **`domain_prefix`**: *(Optional `string`)*
+
+  Type a domain prefix to use for the sign-up and sign-in pages that are hosted by Amazon Cognito,
+  e.g. `https://{YOUR_PREFIX}.auth.eu-west-1.amazoncognito.com`. The prefix must
+  be unique across the selected AWS Region. Domain names can only contain
+  lower-case letters, numbers, and hyphens. Default is `""`.
```
Comment (mariux) [inline README.md:359]: this is only one use case... if no certificate arn is givin or this is not a fqdn ... also this argument should be domain

File: main.tf [hunk 3]
```diff
@@ -201,3 +201,11 @@ resource "aws_cognito_user_pool_client" "client" {
...
+  certificate_arn = var.domain_certificate_arn
```
Comment (mariux) [inline main.tf:209]:
```suggestion
  certificate_arn = var.certificate_arn
```

File: README.md [hunk 4]
```diff
@@ -350,6 +349,20 @@ for details and use-cases.
...
+
+- **`domain_certificate_arn `**: *(Optional `string`)*
+
+  The ARN of an ISSUED ACM certificate in us-east-1 for a custom domain.
+  Default is `null`.
```
Comment (mariux) [inline README.md:364]: null should not be communicated to the user... "default is not to use a custom domain"

Comment (soerenmartius) [reply README.md]: resolved in 3b4e65b

File: main.tf (hunk 3 above)
Comment (soerenmartius) [reply main.tf]: resolved in 3b4e65b

File: README.md (hunk 2 above)
Comment (soerenmartius) [reply README.md]: resolved in 3b4e65b

File: main.tf (hunk 1 above)
Comment (soerenmartius) [reply main.tf]: resolved in 3b4e65b

File: README.md [hunk 5]
```diff
@@ -48,7 +48,6 @@ pre-configured.
   Create Cognito User Pool Clients.
 
 - *Features not yet implemented*:
-  [`cognito_user_pool_domain`](https://www.terraform.io/docs/providers/aws/r/cognito_user_pool_domain.html)
```
Comment (mariux) [inline README.md:51]: need to be moved to implemented not just deleted ;)

File: README.md [hunk 6]
```diff
@@ -350,6 +349,23 @@ for details and use-cases.
...
+- **`domain`**: *(Optional `string`)*
+
+  The domain name that should be used. Can be set to a FQDN or prefix.
+  If no FQDN and `certificate_arn` are set, the domain prefix will be used for
+  the sign-up and sign-in pages that are hosted by Amazon Cognito,
+  e.g. `https://{YOUR_PREFIX}.auth.eu-west-1.amazoncognito.com`.
+  The prefix must be unique across the selected AWS Region.
+  Domain names can only contain lower-case letters, numbers, and hyphens. 
+  
+  Default is `null`.
```
Comment (mariux) [inline README.md:363]: default null should not be communicated to users in docs..

File: README.md [hunk 7]
```diff
@@ -350,6 +349,23 @@ for details and use-cases.
...
+
+- **`certificate_arn `**: *(Optional `string`)*
+
+  The ARN of an ISSUED ACM certificate in us-east-1 for a custom domain.
+  Default is `default is not to use a custom domain`.
```
Comment (mariux) [inline README.md:368]:
```suggestion
  Default is not to use a custom domain.
```

File: README.md (hunk 5 above)
Comment (soerenmartius) [reply README.md:51]: resolved in 28d556a

File: README.md (hunk 6 above)
Comment (soerenmartius) [reply README.md]: resolved in 28d556a

File: README.md (hunk 7 above)
Comment (soerenmartius) [reply README.md]: resolved in 28d556a

**Pull request merged**